packages/website/lib/api.js

```diff
@@ -307,15 +307,17 @@ export async function deletePinRequest(requestid) {
 
 /**
  * Gets/Puts saved user plan and billing settings.
+ * @param {string} agreement
```

Comment:
Can we type this more specifically like you did in the api with a well-known list of agreement values? I think it will really help avoid bugs getting to prod because we'll be able to check things at compile time.

I don't really like how this function sometimes does a GET and sometimes does a PUT, but didn't want to nitpick at the time. But I think in both cases it's actually fine to allow this to be undefined. e.g. you might wanna use the API just to save a payment method, but without recording an agreement or subscription price.

Reply:
I hadn't had time yet but it seems like we should have a new small package in the monorepo for things that are shared across the API and the website, e.g. the known agreement types.

It would make it easier to type these kinds of things in sync. For now since we only have one agreement type I might just do
```suggestion
 * @param {'web3.storage-tos-v1'} agreement
```

(if anyone has time though we should go ahead and add the new package, but it takes a bit wrestling with the monorepo linking sometimes)

Reply:
I typed this more explicitly (I think in https://github.com/web3-storage/web3.storage/pull/2028/commits/eb506e338c403f970ce0bb6c19fcc35d92ea04ec)